scipy/optimize/_trustregion.py

```diff
@@ -210,7 +214,12 @@ def hessp(x, p, *args):
     x = x0
     if return_all:
         allvecs = [x]
-    m = subproblem(x, fun, jac, hess, hessp)
+
+    subproblem_init_kw = {}
+    if hasattr(subproblem, 'MAXITER_DEFAULT'):
```

Comment:
This was the least intrusive way I could think of to extend the QuadraticSubproblem class. Either the extra option would go into the `__init__` or the `solve` method. Here I chose to pass it in __init__. Another option would be to have a generic `subproblem_options` dictionary passed to all constructors (and warn when unused options are passed?), but the call to minimize would then look like: `minimize(fun, x0, jac, options={"gtol": 1e-6, "subproblem_options": {'maxiter': 10}}`. But perhaps that's preferable. Let me know what you think.

Reply:
I think this looks okay as is, but I am not a heavy user of `minimize`, so it would be good for someone else to weigh in.